Library/include/CSP/Common/Optional.h

```diff
@@ -71,17 +71,6 @@ template <typename T> class CSP_API Optional
 		ValueDestructor = DefaultDestructor<T>;
 	}
 
-	/// @brief Constructs an optional with a given value of a different type.
-	/// @param InValue const U* : Pointer to construct optional with
-	template <typename U> Optional(const U* InValue)
```

Comment:
This constructor seems to actually be redundant as it is covered by the one below `template <typename U> Optional(const U& InValue)`. This is tested with the `OptionalWithExtendedClassPointerInitialisationTest` test where `U` is `OptionalExtraTestClass*` as opposed to `OptionalWithExtendedClassReferenceInitialisationTest` where `U` is interpreted as `OptionalExtraTestClass`.

Reply:
This constructor is here to allow creation of an `Optional<T>` instance from a `U*`. The reason `OptionalWithExtendedClassPointerInitialisationTest` passes is due to `OptionalExtraTestClass` deriving from `OptionalTestClass`, so the `OptionalTestClass(OptionalTestClass* Other)` constructor is used.

Reply:
I did add `OptionalTestClass(OptionalTestClass* Other)` to enable this test to pass. From what I can tell though, the path the code takes is through `template <typename U> Optional(const U& InValue)` where `U` is an `OptionalTestClass*`. I am unsure how I can force it to use the `template <typename U> Optional(const U& InValue)` constructor. Will take another look with unrelated classes.

Reply:
I have tried a bunch of ways of testing this but I have not found any case where `template <typename U> Optional(const U* InValue)` is used. Every time `template <typename U> Optional(const U& InValue)` is used. I do not believe the code I removed has any impact. If you can provide a test that would utilise it I'm happy to try it out but otherwise I think we should leave it removed.

Reply:
If your class has a constructor that takes a `const uint32_t*` and you construct an Optional using a value of this type, it will use the `Optional(const U* InValue)` constructor and not the `const U&` one. This constructor needs to stay in order to support this.

Example code showing this: https://onlinegdb.com/OKSrbiODT

Reply:
Thanks for the details. Will give it a go adding a test like this.

Reply:
Having tried this there seems to be no change. Any tests I try pass fine without undeleting this constructor. 

Having chatted a bit offline are you happy to resolve this @MAG-MichaelK ?